[Weiye] iP

TeddYE opened the pull request.

TeddYE committed:
Duke class: implement level 1 functionalities

Let's implement a skeletal version of Duke that starts by greeting
the user, simply echoes commands entered by the user, and exits when
the user types bye.

TeddYE committed:
Duke class: Implement level 2 functionalities

Let's add the ability to store whatever text entered by the user
and display them back to the user when requested.

TeddYE committed:
Duke class: remove redundant codes

Let's,
* move the conditional break in the infinite loop into the loop
condition
* remove String.format()

TeddYE committed:
Duke class: Implement Level-3 functionalities

Lets,
* add Task class to represent tasks
* add the ability to mark tasks as done and the ability to change
the status back to not done

TeddYE committed:
Duke class: implement Level-4 functionalities

As there are multiple types of tasks that have some similarity
between them, implement classes Todo, Deadline and Event classes
to inherit from a Task class.

Let's,
* add support for tracking three types of task

TeddYE committed:
Update i/o files for testing

Use the input/output redirection technique to semi-automate the
testing of Duke.

TeddYE committed:
Duke class: implement level-5 functionalities

Teach Duke to deal with errors such as incorrect inputs entered by
the user.

Let's
* add DukeException class

TeddYE committed:
Duke class: implement level-6 functionalities

Let's
* Add support for deleting tasks from the list

TeddYE committed:
Implement A-Enums functionalities

Let's:
* use enum for types of tasks

TeddYE committed:
Add parser class to parse the user input into different commands

Let's:
* add abstract class for different commands

TeddYE committed:
Duke class: implement level-7 functionalities

Let's:
* add Storage class to facilitate reading and writing of list to file

TeddYE committed:
Merge pull request #1 from TeddYE/branch-Level-7

Branch level 7

TeddYE committed:
Implement level-8 functionalities

Teach duke to understand date and time.

Let's
* update due date and event time to LocalDateTime instead of String

TeddYE committed:
Merge pull request #2 from TeddYE/branch-Level-8

Implement level-8 functionalities

TeddYE committed:
Refactor the code to extract out closely related code as classes

Let's:
* add TaskList class
* add Ui class

TeddYE committed:
Merge pull request #3 from TeddYE/branch-Level-8

A-MoreOOP

TeddYE committed:
Organize the classes into suitable java packages

TeddYE committed:
Add JUnit tests to test the behavior of the code

TeddYE committed:
Package the app as an executable JAR file

TeddYE committed:
Add JavaDoc comments to the code

TeddYE committed:
Reorganize the directories

TeddYE committed:
Merge pull request #4 from TeddYE/branch-A-JavaDoc

Add JavaDoc comments to the code

TeddYE committed:
Merge branch 'master' of https://github.com/TeddYE/ip

# Conflicts:
#	src/main/META-INF/MANIFEST.MF

TeddYE committed:
Implement Level-9 functionalities

Let's:
* add CFind class to handle the search function
* add appropriate response after the search is done

TeddYE committed:
Merge pull request #6 from TeddYE/branch-Level-9

Implement Level-9 functionalities

hieunm1821 commented on src/main/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+        this.parser = new Parser();
+    }
+
+    public void run() throws DukeException{
```
I think it should be a white space before the open curly bracket. I noticed the same issue in several other places too.

hieunm1821 commented on src/main/duke/TaskList.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> Tasks;
```
I think the variable name should be in camelCase. I suggest it should be _tasks_

hieunm1821 commented on src/main/duke/commands/CBye.java:
```diff
@@ -0,0 +1,18 @@
+import main.duke.Ui;
+import main.duke.enums.CommandType;
+
+public class CBye extends Command{
```
Should the name of class be CommandBye? I think it is more meaningful. Similar with other command names.

hieunm1821 commented on src/main/duke/commands/CFind.java:
```diff
@@ -0,0 +1,32 @@
+        this.findString = findString;
+    }
+
+    public String getFindString() { return this.findString; }
```
I suggest follow EgyptianStyle when we make a function which has one line only. I noticed the same issue in several other places too.

hieunm1821 commented on src/main/duke/io/Parser.java:
```diff
@@ -0,0 +1,121 @@
+                        // check if the date and time input is in the right format
+                        if (deadlineDescription.equals("") || dueDate.equals("")) {
+                            throw new DukeException("Please specify the description/due date of the deadline task.");
+                        }
+                        newCommand = new CDeadline(deadlineDescription, dueDate);
+                    } catch (DateTimeParseException e){
+                        throw new DukeException("Please provide the date time in this format YYYY-MM-DD 0000");
+                    }
+                }
+                break;
+            case "event":
+                if (!userInput.contains("/at")) {
+                    throw new DukeException("Please specify the date time using the /at keyword.");
+                } else {
+                    try {
+                        int byIndex = Arrays.asList(inputArray).indexOf("/at");
+                        String eventDescription = String.join(" ",
+                                Arrays.copyOfRange(inputArray, 1, byIndex));
+                        String dateTime = String.join(" ",
+                                Arrays.copyOfRange(inputArray, byIndex + 1, inputArray.length));
+                        LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd kkmm"));
+                        // check if the date and time input is in the right format
+                        if (eventDescription.equals("") || dateTime.equals("")) {
+                            throw new DukeException("Please specify the description/date time of the event task.");
+                        }
+                        newCommand = new CEvent(eventDescription, dateTime);
+                    } catch (DateTimeParseException e){
+                        throw new DukeException("Please provide the date time in this format YYYY-MM-DD 0000");
+                    }
+                }
+                break;
+            case "find":
+                String findString = String.join(" ",
+                        Arrays.copyOfRange(inputArray, 1, inputArray.length));
+                newCommand = new CFind(findString);
+                break;
+            default:
+                throw new DukeException("Sorry. I do not understand your input.");
+        }
+        return newCommand;
```
I think there is no indentation for _case_ clauses. I suggest you change the setting in the IDE.

hieunm1821 commented on src/main/duke/tasks/ToDo.java:
```diff
@@ -0,0 +1,13 @@
+
+import main.duke.enums.TaskType;
+
+public class ToDo extends Task{
```
Should the class name Todo? :thinking:

hieunm1821 reviewed (commented):
Overall, I found your code easy to read. However, there are a few minor coding standard. Just a few nits to fix.

TeddYE committed:
Use Gradle to automate some of the build tasks of the project

TeddYE committed:
Merge pull request #7 from TeddYE/add-gradle-support

Use Gradle to automate some of the build tasks of the project

kahleongq commented on src/main/duke/io/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        String userCommand = inputArray[0];
+        Command newCommand;
+
+        switch (userCommand) {
```
there should not be an indentation for case clauses to follow the coding standards

kahleongq reviewed (commented):
Overall you adhered mostly to the coding standards!

TeddYE committed:
Implement Level-10 functionalities

Add GUI supoport

TeddYE committed:
Merge pull request #8 from TeddYE/branch-Level-10

Implement Level-10 functionalities

TeddYE committed:
Improve code quality based on Code Quality topics

TeddYE committed:
Merge pull request #9 from TeddYE/branch-A-CodeQuality

Improve code quality based on Code Quality topics

TeddYE committed:
Use Assertions

TeddYE committed:
Merge pull request #10 from TeddYE/branch-A-Assertions

Use Assertions

TeddYE committed:
Implement undo command for users

Added cloning function for the tasks so that the previous state of
the tasks list can be saved.

Stack is used to store the previous state of the tasks list.

Unifying all those duplicated code improves the code quality.

As a step towards such unification, let's extract those duplicated
code blocks into separate methods in their respective classes.
Doing so will make the subsequent unification easier.

TeddYE committed:
Add user guide

TeddYE committed:
Merge pull request #12 from TeddYE/branch-UserGuide

Add user guide

TeddYE committed:
Set theme jekyll-theme-minimal

TeddYE committed:
Set theme jekyll-theme-dinky

TeddYE committed:
Update comments in accordance to coding standards

TeddYE committed:
Add ui picture

TeddYE committed:
Set theme jekyll-theme-dinky

TeddYE committed:
Change ui picture file name

TeddYE committed:
Merge branch 'master' of https://github.com/TeddYE/ip

TeddYE committed:
Update main class name in build.gradle
